src/pytorch_lightning/cli.py

```diff
@@ -472,11 +472,13 @@ def _add_subcommands(self, parser: LightningArgumentParser, **kwargs: Any) -> No
         )
         # register all subcommands in separate subcommand parsers under the main parser
         for subcommand in self.subcommands():
-            subcommand_parser = self._prepare_subcommand_parser(trainer_class, subcommand, **kwargs.get(subcommand, {}))
-            self._subcommand_parsers[subcommand] = subcommand_parser
             fn = getattr(trainer_class, subcommand)
             # extract the first line description in the docstring for the subcommand help message
             description = _get_short_description(fn)
+            subparser_kwargs = kwargs.get(subcommand, {})
+            subparser_kwargs["description"] = description
```

Comment:
Should we worry about not overwriting?

```suggestion
            subparser_kwargs.setdefault("description", description)
```